src/osp/Active/SysVehicleSync.cpp

```diff
@@ -0,0 +1,239 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2021 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "SysVehicleSync.h"
+
+#include "ActiveScene.h"
+#include "SysAreaAssociate.h"
+#include "SysHierarchy.h"
+#include "SysPhysics.h"
+#include "SysVehicle.h"
+#include "drawing.h"
+
+#include "../Satellites/SatActiveArea.h"
+#include "../Satellites/SatVehicle.h"
+
+using namespace osp::active;
+
+using osp::universe::Universe;
+using osp::universe::Satellite;
+
+using osp::universe::CoordinateSpace;
+
+using osp::universe::UCompActiveArea;
+using osp::universe::UCompVehicle;
+
+// for the 0xrrggbb_rgbf literalsm
+using namespace Magnum::Math::Literals;
+
+ActiveEnt SysVehicleSync::activate(ActiveScene &rScene, Universe &rUni,
+                                   Satellite areaSat, Satellite tgtSat)
+{
+    SPDLOG_LOGGER_INFO(rScene.get_application().get_logger(),
+                      "Loading a vehicle");
+
+    auto &loadMeVehicle = rUni.get_reg().get<universe::UCompVehicle>(tgtSat);
+    auto &tgtPosTraj = rUni.get_reg().get<universe::UCompTransformTraj>(tgtSat);
+
+    // make sure there is vehicle data to load
```

Comment:
Thoughts on prefixing std::optional with m_oBlueprint, similar to m_rBlueprint for references or m_pBlueprint?

Reply:
I'm cool with that; guessing every kind of single-element 'container' should do this? It doesn't seem that useful for local variables however, since an explicit std::optional<...> type declaration would already be present and easy to spot.

This case with m_blueprint is actually a DependRes and not an optional though.